plugins/inputs/couchdb/couchdb.go

```diff
@@ -3,65 +3,96 @@ package couchdb
 import (
 	"encoding/json"
 	"fmt"
-	"github.com/influxdata/telegraf"
-	"github.com/influxdata/telegraf/plugins/inputs"
 	"net/http"
 	"reflect"
 	"sync"
 	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
-// Schema:
 type metaData struct {
-	Description string  `json:"description"`
-	Current     float64 `json:"current"`
-	Sum         float64 `json:"sum"`
-	Mean        float64 `json:"mean"`
-	Stddev      float64 `json:"stddev"`
-	Min         float64 `json:"min"`
-	Max         float64 `json:"max"`
+	Current JSONFloat64 `json:"current"`
+	Sum     JSONFloat64 `json:"sum"`
+	Mean    JSONFloat64 `json:"mean"`
+	Stddev  JSONFloat64 `json:"stddev"`
+	Min     JSONFloat64 `json:"min"`
+	Max     JSONFloat64 `json:"max"`
+	Value   JSONFloat64 `json:"value"`
+}
+
+type JSONFloat64 float64
+
+// Used to restore Couchdb <2.0 API behavior
+func (j *JSONFloat64) UnmarshalJSON(data []byte) error {
```

Comment:
Is this for the unset vs 0 value support?  I think this can be done simpler by making each field in metaData a `*float64`.

Reply:
I did try it with the `*float64`, in pre 2.0, the JSON response would contain `{"key":null}` but the plugin would set it to 0. Since we don't wan't to set every [empty] key to 0 (as the 2.0 version would be spammed with unset fields), nor use the pointer and have it not parse "null" values in <2, this `UnmarshalJSON` function is needed to preserve the current behavior.